youtube_downloader/downloader.py

```
@@ -43,8 +97,46 @@ def _get_video_info(self):
         }
         
-        response = self.session.post(api_url, json=payload)
-        return response.json()
+        for attempt in range(retries):
+            try:
+                response = self.session.post(api_url, json=payload, timeout=30)
+                
+                # Handle rate limiting
+                if response.status_code == 429:
+                    if self.proxy_manager:
+                        print(f"⚠ Rate limited (429). Rotating proxy...")
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)